examples/virtio-snd/snd_driver_vmm.c

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright 2023, UNSW
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+#include <stddef.h>
+#include <stdint.h>
+#include <microkit.h>
+#include "util/util.h"
+#include "arch/aarch64/vgic/vgic.h"
+#include "arch/aarch64/linux.h"
+#include "arch/aarch64/fault.h"
+#include "guest.h"
+#include "fault.h"
+#include "virq.h"
+#include "tcb.h"
+#include "vcpu.h"
+#include "virtio/console.h"
+#include <sddf/serial/queue.h>
+#include <sddf/sound/queue.h>
+#include <sddf/util/cache.h>
+#include <uio/sound.h>
+
+/*
+ * As this is just an example, for simplicity we just make the size of the
+ * guest's "RAM" the same for all platforms. For just booting Linux with a
+ * simple user-space, 0x10000000 bytes (256MB) is plenty.
+ */
+#define GUEST_RAM_SIZE 0x10000000
+
+#if defined(BOARD_qemu_arm_virt)
+#define GUEST_DTB_VADDR 0x47000000
+#define GUEST_INIT_RAM_DISK_VADDR 0x46000000
+#elif defined(BOARD_odroidc4)
+#define GUEST_DTB_VADDR 0x27000000
+#define GUEST_INIT_RAM_DISK_VADDR 0x26000000
+#else
+#error Need to define guest kernel image address and DTB address
+#endif
+
+/* Data for the guest's kernel image. */
+extern char _guest_kernel_image[];
+extern char _guest_kernel_image_end[];
+/* Data for the device tree to be passed to the kernel. */
+extern char _guest_dtb_image[];
+extern char _guest_dtb_image_end[];
+/* Data for the initial RAM disk to be passed to the kernel. */
+extern char _guest_initrd_image[];
+extern char _guest_initrd_image_end[];
+/* Microkit will set this variable to the start of the guest RAM memory region. */
+uintptr_t guest_ram_vaddr;
+
+#define SND_CLIENT_CH 4
+
+// @alexbr: why 50?
+#define UIO_SND_IRQ 50
+
+#define MAX_IRQ_CH 63
+int passthrough_irq_map[MAX_IRQ_CH];
+
+
+#define SERIAL_MUX_TX_CH 1
+#define SERIAL_MUX_RX_CH 2
```

Comment:
```suggestion
#define SERIAL_TX_CH 1
#define SERIAL_RX_CH 2
```

Should refer to virt but it's easier to just refer to the RX and TX directions, the client doesn't care what's on the other end (if it's some other component or an actual virtualiser/mux)

Reply:
Done in https://github.com/au-ts/libvmm/commit/772ca2a58cf66ed4e022b4fdad508b8c00930215